nncf/common/pruning/schedulers.py

```diff
@@ -108,6 +109,11 @@ def current_pruning_level(self) -> float:
             return self._current_level
         return 0
 
+    def load_state(self, state: Dict[str, Any]) -> None:
+        super().load_state(state)
+        if self._current_level is None:
+            self._current_level = self._calculate_pruning_level()
```

Comment:
Please consider implementing `get_state` to save `_current_level` and modifying loading from given state, otherwise `self_current_level` always will be `None` on checkpoint loading. 

We should decide should we keep current pruning level is state dict or we should always calculate it (in this case if statement above should be removed)

Reply:
It is preferable to calculate than to store. @daniil-lyakhov, what do you mean under otherwise `self_current_level` always will be `None` on checkpoint loading? I see the opposite:
```
        if self._current_level is None:
            self._current_level = self._calculate_pruning_level()
```

Reply:
I mean, if  `load_state` always will be called before `epoch_step`, `_current_level`  after scheduler initialization always will be `None` in `load_state`. In case one would load model after `epoch_step` call, all scheduler stats will be updated besides `_current_level`, which will be properly recalculated only on `epoch_step`

Reply:
I don't understand the problem.

> I mean, if `load_state` always will be called before `epoch_step`, `_current_level` after scheduler initialization always will be `None` in `load_state`.

Seems correct. If `_current_level` is `None` inside `load_state()` it will be calculated according to `current_epoch` and `current_step` attributes.

> In case one would load model after `epoch_step` call, all scheduler stats will be updated besides `_current_level`, which will be properly recalculated only on `epoch_step`

Seems like it is the wrong case. The model should be initialized before `epoch_step()`.

Reply:
@andrey-churkin, I know this isn't real case, but if one start training from scratch and then, after training, load checkpoint, the model contains `_current_level` from scratch training

Reply:
I suggest always recalculate `_current_level` on `load_model`

Reply:
`_current_level` attr was removed as discussed